src/app/components/framework/custom-popover/custom-popover.component.ts

```diff
@@ -0,0 +1,247 @@
+import { ChangeDetectionStrategy, Component, HostBinding, Input, OnInit, ViewEncapsulation } from '@angular/core';
+
+export interface PopOverOpts {
+  /** Selects the direction the box is 'popped out' towards */
+  direction?: 'above' | 'below' | 'left' | 'right';
+
+  /**
+   * Selects the point to which the pop over element is attached to.
+   * The following attachment points are available:
+   *  (A)--(B)--(C)
+   *   |    |    |
+   *  (D)--(E)--(F)
+   *   |    |    |
+   *  (G)--(H)--(I)
+   */
+  attachmentPoint?: 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I';
+
+  /**
+   * Selects the positioning of the pop over element in reference to the attachment point. Assuming the direction to be:
+   *  - above or below -> 0 = left bound, 1 = centered, 2 = right bound
+   *  - left or right  -> 0 = top bound,  1 = centered, 2 = bottom bound
+   *
+   * Example: attachmentPoint = 'D' ; referencePoint = '0'
+   *
+   *                     (A)--------
+   *                      |
+   *     ---------(0) -> (D) Anchor
+   *               |      |
+   *     Pop over (1)    (G)--------
+   *               |
+   *     ---------(2)
+   */
+  referencePoint?: '0' | '1' | '2';
+}
+
+@Component({
+  selector: 'app-custom-popover',
+  templateUrl: './custom-popover.component.html',
+  styleUrls: ['./custom-popover-internal.component.css', './custom-popover-style.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CustomPopoverComponent implements OnInit {
+  /** (Optional) The provided HTMLElement's CSS property 'position' is set to 'relative' if it is not
+   *  already set to either 'relative' of 'absolute' */
+  @Input() forcedAnchor: HTMLElement;
+
+  /** (Optional) Sets the positioning of the popover box in relation to its parent */
+  @Input() options: PopOverOpts;
+
+  /** (Optional) Weather to show (true) or hide (false) the decorator caret */
+  @Input() showCaret = true;
+
+  /** (Optional) Custom style class applied to the host. This should be chosen from one of the
+   *  classes defined in the 'custom-popover-style.component.css' file. If a desired style is not available,
+   *  it should also be created there. */
+  @Input() customClass = '';
+
+  @HostBinding('className')
+  private get hostClass(): string {
+    return this._hostClass;
+  }
+
+  /** Class values for internal use */
+  _boxClass = 'popover-box';
+  _contentClass = 'popover-content';
+  _caretClass = 'popover-caret';
+  _hostClass;
+
+  ngOnInit(): void {
+    this.forceAnchorStyle();
+    const opts: PopOverOpts =
+      this.options !== undefined ? this.options : { direction: 'above', referencePoint: '1', attachmentPoint: 'B' };
+    this.selectCaret(opts);
+    this._hostClass = this.buildHostClass(opts);
+  }
+
+  private forceAnchorStyle() {
+    if (this.forcedAnchor !== undefined) {
+      if (!(this.forcedAnchor.style.position === 'relative' || this.forcedAnchor.style.position === 'absolute')) {
```

Comment:
würde es nicht sinnvoll sein einen Error zu schmeißen, wenn der Anchor position: fixed; oder position: sticky; hat? static kannst du ja einfach in relative ändern ohne große Auswirkungen(correct me if im wrong), aber bei fixed/sticky verändert das das Erscheinungsbild des Anchors doch erheblich, oder?